radix-engine/src/engine/component_objects.rs

```diff
@@ -87,13 +87,13 @@ impl ComponentObjectRefs {
         // Only allow vaults to be added, never removed
         for vault_id in &other.vault_ids {
             if !self.vault_ids.remove(&vault_id) {
-                return Err(RuntimeError::VaultRemoved(vault_id.clone()));
+                return Err(RuntimeError::VaultRemoved());
```

Comment:
Why are we removing the arguments?

Reply:
Ah, I see. I'd suggest we keep the arguments. We're only going to add more in the long run. They provide useful information for both debugging and testing.  The `matches!` macro can help with loose comparison (although I'd advocate full comparison for testing).